specs/WindowControlsOverlayConfiguration.md

```diff
@@ -0,0 +1,120 @@
+The overlay configuration lives on the controller object.
+
+This API is designed to be used in addition with the other non-client region APIs 
+and features. These include `app-region: drag`, and `IsNonClientRegionSupportEnabled`.
+# Examples
+
+## Win32 C++
+```cpp 
+AppWindow::AppWindow() {
+    m_mainWindow = CreateWindowExW(
+        WS_EX_CONTROLPARENT, 
+        GetWindowClass(),
+        L""
+        WS_POPUPWINDOW, 
+        0,0, 800, 800, 
+        nullptr, nullptr, 
+        g_hInstance, nullptr);
+}
+
+void  AppWindow::OnCreateWebview2ControllerCompleted(HRESULT hr, ICoreWebview2Controller* controller) 
+{
+    wil::com_ptr<ICoreWebView2Controller5> controller5;
+    CHECK_FAILURE(controller->QueryInterface(&controller5));
+
+    wil::com_ptr<ICoreWebView2WindowControlsOverlayConfiguration> wco_config;
+    CHECK_FAILURE(controller5->get_WindowControlsOverlayConfiguration(&wco_config));
+
+    wco_config->put_IsEnabled(true);
+    COREWEBVIEW2_COLOR color {1, 0, 0, 225};
+    wco_config->put_TitleBarColor(color);
+}
+```
+## .NET C#
+```c#
+// WebView2 control is defined in the xaml
+// <wv2:WebView2 x:Name="webView" Source="https://www.microsoft.com/"/>
+public MainWindow() 
+{
+    InitializeComponent();
+    m_AppWindow.TitleBar.ExtendsContentIntoTitleBar = true;
+
+    CoreWebView2WindowControlsOverlayConfiguration config = _coreWebView2Controller.WindowControlsOverlayConfiguration;
+    config.IsEnabled = true; 
+    config.color = Color.FromARGB(0, 0, 255);
+}
+```
+
+# API Details
+## Win32 C++
+```cpp
+/// Controller API used to configure the window controls overlay.
+/// To provide your app users with the best experience, it is important to handle webview 
+/// initialization errors appropriatly. Provide your users with a way to close the window
+/// or restart the App.
+[uuid(101e36ca-7f75-5105-b9be-fea2ba61a2fd), object, pointer_default(unique)]
+interface ICoreWebView2Controller5 : IUnknown {
+  /// Gets the `WindowControlsOverlayConfiguration` object.
+  [propget] HRESULT WindowControlsOverlayConfiguration([out, retval] ICoreWebView2WindowControlsOverlayConfiguration** value);
+}
+
+/// This is the ICoreWebView2WindowControlsOverlayConfiguration
+[uuid(c9f7378b-8dbb-5445-bacb-08a3fdf032f0), object, pointer_default(unique)]
+interface ICoreWebView2WindowControlsOverlayConfiguration : IUnknown {
+  /// Gets the `Height` property.
+  [propget] HRESULT Height([out, retval] UINT32* value);
+
+  /// The `Height` property in pixels, allows devs to set the height of the overlay 
+  /// Defaults to 48px. 
+  [propput] HRESULT Height([in] UINT32 value);
+
+  /// Gets the `IsEnabled` property.
+  [propget] HRESULT IsEnabled([out, retval] BOOL* value);
+
+  /// The `IsEnabled` property allows devs to opt in/out of using
+  /// the WV2 custom caption controls. Defaults to `FALSE`.
+  /// 
+  /// When this property is `TRUE`, WV2 will draw its own caption controls on the
+  /// window.
```

Comment:
It would be good to document:
 * what controls are displayed (min, max/restore, close?)
 * How this interacts with HTML content in the same position (this covers it?)
 * Scenario and other API recommendations: Eg: You should use this when the WebView2 is covering its entire window, the window should be configured to not display its own window controls, use app-region: drag and IsNonClientRegionSupportEnabled somehow?, ensure you don't important HTML rendering under the window controls?, anything else recommended to cover this scenario

Reply:
Included